python-package/lightgbm/basic.py

```diff
@@ -727,6 +727,36 @@ def _free_handle(self):
             self.handle = None
         return self
 
+    def _set_init_score_by_predictor(self, predictor, data, used_indices=None):
+        data_has_header = False
+        if isinstance(data, string_type):
+            # check data has header or not
+            if str(params.get("has_header", "")).lower() == "true" \
+                    or str(params.get("header", "")).lower() == "true":
```

Comment:
I think it's enough to check `if params.get("has_header", False) or params.get("header", False)` because we don't check that other boolean params can be string at Python side.